udaykalra/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -36,14 +36,43 @@ private String arrayListToJson(ArrayList<String> inputList) {
     String json = gson.toJson(inputList);
     return json;
   }
+
+  /**
+  * Obtains comments json to display upon request.
+  */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("text/html;");
-    comments = new ArrayList<String>();
-    comments.add("What's good");
-    comments.add("Howdy friend!");
-    comments.add("Hello homie.");
     String outputJson = arrayListToJson(comments);
     response.getWriter().println(outputJson);
   }
+
+  /**
+  * Writes comment data to ArrayList data
+  */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String text = getParameter(request, "text-input", "");
+    boolean eraser = Boolean.parseBoolean(getParameter(request, "erase", "false"));
+    comments.add(text);
+
+    if (eraser) {
+      comments = new ArrayList<String>();
+    }
+
+    // Redirect back to the HTML page.
+    response.sendRedirect("/index.html");
+  }
+
+  /**
+  * Obtains parameter from Comments typing field.
+  */
+  private String getParameter(HttpServletRequest request, String comment, String defaultValue) {
```

Comment:
Seems like `comment` would be more accurately named `parameter` or `parameterName`, since you're looking for a specific parameter in request (this could be a comment, but could also be the "erase" checkbox).

Reply:
Done